src/CarbonAware.WebApi/src/Controllers/CarbonAwareController.cs

```diff
@@ -278,12 +278,33 @@ public async Task<IActionResult> GetAverageCarbonIntensity([FromQuery, BindRequi
     [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(ValidationProblemDetails))]
     [ProducesResponseType(StatusCodes.Status500InternalServerError, Type = typeof(ValidationProblemDetails))]
     [HttpPost("average-carbon-intensity/batch")]
-    public IActionResult GetAverageCarbonIntensityBatch(IEnumerable<CarbonIntensityBatchDTO> requestedCarbonIntensities)
+    public async IAsyncEnumerable<CarbonIntensityDTO> GetAverageCarbonIntensityBatch(IEnumerable<CarbonIntensityBatchDTO> requestedCarbonIntensities)
```

Comment:
You need to change the comment of the 200k response (line 277) since now you're returning `CarbonIntensityDTO` instances. Or you just remove the `Type = ...` from the 200 response and swagger takes care of it. (preferred)

Reply:
agreed